eclair-core/src/main/scala/fr/acinq/eclair/payment/CommandBuffer.scala

```diff
@@ -29,44 +31,49 @@ import fr.acinq.eclair.channel._
 class CommandBuffer(nodeParams: NodeParams, register: ActorRef) extends Actor with ActorLogging {
 
   import CommandBuffer._
-  import nodeParams.db._
 
   context.system.eventStream.subscribe(self, classOf[ChannelStateChanged])
 
   override def receive: Receive = {
+    val pendingRelays = new mutable.HashMap[ByteVector32, mutable.Set[HasHtlcIdCommand]] with mutable.MultiMap[ByteVector32, HasHtlcIdCommand]
+    for ((chanId, cmd) <- nodeParams.db.pendingRelay.listPendingRelay()) pendingRelays.addBinding(chanId, cmd)
+    main(pendingRelays)
+  }
+
+  def main(pendingRelays: mutable.HashMap[ByteVector32, mutable.Set[HasHtlcIdCommand]] with mutable.MultiMap[ByteVector32, HasHtlcIdCommand]): Receive = {
 
-    case CommandSend(channelId, htlcId, cmd) =>
+    case CommandSend(channelId, cmd) =>
       // save command in db
-      register forward Register.Forward(channelId, cmd)
+      register ! Register.Forward(channelId, cmd)
       // we also store the preimage in a db (note that this happens *after* forwarding the fulfill to the channel, so we don't add latency)
-      pendingRelay.addPendingRelay(channelId, htlcId, cmd)
+      nodeParams.db.pendingRelay.addPendingRelay(channelId, cmd)
+      context become main(pendingRelays.addBinding(channelId, cmd.modify(_.commit).setTo(false)))
```

Comment:
It's because of this line: https://github.com/ACINQ/eclair/pull/1111/files#diff-5a1f33d183f796f6859fb09367579dafR64

Once stashed they all should have `commit = false` because `CommandBuffer` sends all of them and then a final `CMD_SIGN`, it's faster this way.

Reply:
Right, then i think this should be in a comment maybe above the declaration of the `CommandBuffer` class. A more fancy solution would be to have a strategy pattern approach to be able to select the desired behavior of the buffer but probably we can stick with this default strategy (but it's nice to leave a comment).